### PR TITLE
Allow featured images to be synced correctly

### LINKED ISSUE #1: Featured Image missing from post sync
Just thought I'd leave a first issue for you :)

### PR DESCRIPTION
**Pull request by benjaminmueller**
Fixes #1

### CODE REVIEW

**Review by themarcusbattle — APPROVED** [review]
This is good go thanks!

**Review by themarcusbattle — APPROVED** [review]

File: includes/class-api.php [hunk 1]
```diff
@@ -194,7 +194,13 @@ public function insert_new_media( $request, $return_local = false ) {
 	    require_once( ABSPATH . '/wp-admin/includes/media.php' );
 
 	    $attachment_args = $request->get_params();
-		$attachment_url = $attachment_args['attachment_url'];
+
+	    // Attachment URL does not exist so bail early.
+	    if ( ! array_key_exists( 'attachment_url', $attachment_args ) ) {
+	    	wp_send_json_error( $data );
```
Comment (themarcusbattle) [inline includes/class-api.php]: Overall this commit is good. It should just use the new return with a check for returning locally. 

return ( $return_local ) ? $data : wp_send_json_error( $data );

File: includes/class-api.php [hunk 2]
```diff
@@ -234,7 +234,7 @@ public function insert_new_media( $request, $return_local = false ) {
 
 	    $data['id'] = $attachment_id;
 
-		return wp_send_json_success( $data );
+		return $data;
```
Comment (themarcusbattle) [inline includes/class-api.php]: This should use the new return with a check for local returns:

return ( $return_local ) ? $data : wp_send_json_error( $data );

File: includes/class-api.php [hunk 3]
```diff
@@ -349,6 +349,9 @@ public function attach_featured_image( $post_id, $post_args ) {
 			return false;
 		}
 
+		// Allow download_url() to use an external host to retrieve featured image.
+		add_filter( 'http_request_host_is_external', array( $this, 'allow_sync_external_host' ), 10, 3 );
```
Comment (themarcusbattle) [inline includes/class-api.php]: Hey @benjaminmueller could you just let me know what this is for and how it works?

File: includes/class-api.php [hunk 4]
```diff
@@ -358,6 +361,13 @@ public function attach_featured_image( $post_id, $post_args ) {
 
 		$response = set_post_thumbnail( $post_id, $thumbnail_id );
 
+		// Remove filter that allowed download_url() to use an external host.
+		remove_filter( 'http_request_host_is_external', array( $this, 'allow_sync_external_host' ) );
+
+	}
+
+	public function allow_sync_external_host( $allow, $host, $url ) {
```
Comment (themarcusbattle) [inline includes/class-api.php]: @benjaminmueller you can just add a doc-block here to explain how it works. Also why does it just return true? Should it always return true by default?

**Review by themarcusbattle — CHANGES_REQUESTED** [review]

File: includes/class-api.php (hunk 1 above)
Comment (benjaminmueller) [reply includes/class-api.php]: This should now be fixed.

File: includes/class-api.php (hunk 3 above)
Comment (benjaminmueller) [reply includes/class-api.php]: We need to return true to this filter to allow an external request to be made when we call [download_url()](https://github.com/themarcusbattle/press-sync/blob/master/includes/class-api.php#L221) this filter defaults to false and this causes an error to be returned when we attempt to download the featured image.

Filter Info:
https://developer.wordpress.org/reference/hooks/http_request_host_is_external/

File: includes/class-api.php (hunk 4 above)
Comment (benjaminmueller) [reply includes/class-api.php]: This filter usually returns false but we need it to return true only while we are attempting to download the featured image. After the image has been downloaded then we can return it to its previous setting.

File: includes/class-api.php (hunk 2 above)
Comment (benjaminmueller) [reply includes/class-api.php]: This should be fixed now.

**Review by themarcusbattle — APPROVED** [review]

**Comment by themarcusbattle** [comment]
Done! Thanks @benjaminmueller

**Pull request merged**